File: backend/JumpIn.Services/Database/AdImage.cs
namespace JumpIn.Services.Database
{
    public class AdImage
    {
        public Guid Id { get; set; }
        public string ImageUrl { get; set; }
        public bool IsMainImage { get; set; }
        public int DisplayOrder { get; set; }
        public DateTime CreatedAt { get; set; }

        // Foreign key
        public Guid AdId { get; set; }
        public virtual Ad Ad { get; set; }
    }
}

File: frontend/jumpin_mobile/lib/models/ad_image.dart
class AdImage {
  final String id;
  final String imageUrl;
  final bool isMainImage;
  final int displayOrder;
  final String adId;

  AdImage({
    required this.id,
    required this.imageUrl,
    this.isMainImage = false,
    this.displayOrder = 0,
    required this.adId,
  });

  factory AdImage.fromJson(Map<String, dynamic> json) {
    return AdImage(
      id: json['id'].toString(),
      imageUrl: json['imageUrl'] ?? '',
      isMainImage: json['isMainImage'] ?? false,
      displayOrder: json['displayOrder'] ?? 0,
      adId: json['adId'].toString(),
    );
  }
}
